private/bufpkg/bufanalysis/print.go

```diff
@@ -305,6 +327,73 @@ func newExternalFileAnnotation(f FileAnnotation) externalFileAnnotation {
 	}
 }
 
+// externalGitLabCodeQualityIssue represents the GitLab Code Quality Report Issue structure
+// expected for the GitLab Code Quality Report output.
+//
+// https://docs.gitlab.com/ci/testing/code_quality/#code-quality-report-format
+// Note that all fields are required for GitLabCodeQualityIssues. If a field is missing,
+// for example, the path of a deleted file in breaking changes, no GitLab Code Quality violation
+// will be surfaced in the Code Quality Report.
+type externalGitLabCodeQualityIssue struct {
+	// Description is a human readable description of the code quality violation. This maps
+	// to the Message in FileAnnotation.
+	Description string `json:"description,omitempty"`
+	// CheckName is a human readable name of the check or rule. This maps to the Type in
+	// FileAnnotation.
+	CheckName string `json:"check_name,omitempty"`
+	// Fingerprint is a unique identifier for the specific code quality violation. This maps
+	// to a shake256 digest of the JSON FileAnnotation.
```

Comment:
Using the `shake256` digest of the JSON `FileAnnotation` for the hash, but open to suggestions on this!